system/services/email/EmailTemplateService.cfc

```diff
@@ -1202,15 +1208,47 @@ component {
 			});
 		}
 
-		var clickStats    = StructNew( "ordered" );
-		var rawClickStats = $getPresideObject( "email_template_send_log_activity" ).selectData(
-			  filter       = { "message.email_template"=arguments.templateId }
-			, selectFields = [ "count( 1 ) as click_count", "link", "link_title", "link_body" ]
-			, extraFilters = extraFilters
-			, autoGroupBy  = true
-			, orderBy      = "click_count desc"
+		var subQuerySelectFields      = [ "count(1) as click_count", "id" ];
+		var rawClickStatsSelectfields = [ "sendLog.click_count", "link", "link_title", "link_body" ];
+		var subQueryGroupBy           = "link_hash,link_title_hash,link_body_hash";
+		var rawClickStatsGroupBy      = "link_hash,link_title_hash,link_body_hash";
+		var dbInfo                    = _getDbInfoService().getDatabaseVersion( "preside" );
```

Comment:
While we do expect that most, if not all, preside apps have a DSN named "preside", it is not a requirement and we do not hardcode this anywhere else in core.

The general approach here is to use our db adapter pattern so:

```cfc
var adapter = $getPresideObject( "email_template_send_log_activity" ).getDbAdapter();
```

We then can/should use that adapter to help generate the SQL we need. In this case, it isn't going to have the methods and so we might need to add some. e.g.

```cfc
if ( !adapter.supportsGroupByWithAnyFields() ) { // rather than checking whether or not SQL server
   // prepare alternative sql
}
```

We can then add that method to the base adapter (returning true) and then have it return false in the SQL server adapters.

Reply:
Have added helper "supportsGroupByAnyField" on baseAdaptor.
Default is false as realised that Postgres does not support the groupBy with any fields either. MySql would return only if the sql_mode does not have "ONLY_FULL_GROUP_BY" .

Additionally, added another helper "supportsGroupByTextField" as MSSQL does not allow group by text field